addons/misra.py

```diff
@@ -1690,8 +1690,9 @@ def misra_12_3(self, data, rawTokens, filename):
                 continue
             nt = v.nameToken
             if nt and nt.scope and nt.scope.type not in ('Enum', 'Class', 'Struct'):
-                name_tokens_map.setdefault(nt.linenr, set())
-                name_tokens_map[nt.linenr].add(nt.column)
+                if nt.file == filename:
```

Comment:
does this mean that we won't write such warnings in the included header files?

Reply:
Yes, that's right.
It's a bit tricky to implement handling the headers. What if we include the same header with MISRA violation in multiple files? In this case, the addon will emit extra warnings for each dump file.

Reply:
.. so well I understand that it will be a bit tricky to match the filename also properly. But you do not need to worry about emitting duplicates right now imho.

Reply:
> As far as I know misra addon removes duplicates (see existing_violations).

Yes, it will work. Thank you.

I found another problem. It seems that `rawTokens` in dump file do not contain raw token entries from the headers. We have regular tokens from headers in `tokensList`, but not raw ones.

Is this deliberate?

The raw tokens fills [here](https://github.com/danmar/cppcheck/tree/878bce99fef4d6b4642e61eece4dd8b3f0853939/lib/cppcheck.cpp#L555) using simplecpp.

Reply:
hmm.. yes that is intentional unfortunately. before preprocessing you can't know what files are included. Imagine such common code:

    #ifdef _WIN32
    #include "a.h"
    # else
    #include "b.h"
    #endif